internal/transform/ledger.go

```diff
@@ -53,27 +50,16 @@ func TransformLedger(inputLedgerMeta xdr.LedgerCloseMeta) (LedgerOutput, error)
 		return LedgerOutput{}, fmt.Errorf("The fee pool (%d) is negative for ledger %d", outputFeePool, outputSequence)
 	}
 
-	outputBaseFee := int32(ledgerHeader.BaseFee)
-	if outputBaseFee < 0 {
-		return LedgerOutput{}, fmt.Errorf("The base fee (%d) is negative for ledger %d", outputBaseFee, outputSequence)
-	}
+	outputBaseFee := uint32(ledgerHeader.BaseFee)
 
-	outputBaseReserve := int32(ledgerHeader.BaseReserve)
-	if outputBaseReserve < 0 {
-		return LedgerOutput{}, fmt.Errorf("The base reserve (%d) is negative for ledger %d", outputBaseReserve, outputSequence)
-	}
+	outputBaseReserve := uint32(ledgerHeader.BaseReserve)
 
-	outputMaxTxSetSize := int32(ledgerHeader.MaxTxSetSize)
-	if outputMaxTxSetSize < 0 {
-		return LedgerOutput{}, fmt.Errorf("The max transaction set size (%d) is negative for ledger %d", outputMaxTxSetSize, outputSequence)
-	} else if outputMaxTxSetSize < outputTransactionCount {
+	outputMaxTxSetSize := uint32(ledgerHeader.MaxTxSetSize)
+	if int64(outputMaxTxSetSize) < int64(outputTransactionCount) {
```

Comment:
Why the cast here for value comparison?

Reply:
outputMaxTxSetSize is a uint32 and outputTransactionCount is an int32. By casting to an int64, both of their respective data ranges are captured and they can be compared safely.